sklearn/tests/test_discriminant_analysis.py

```diff
@@ -95,6 +97,73 @@ def test_lda_predict():
     assert_raises(ValueError, clf.fit, X, y)
 
 
+@pytest.mark.parametrize("n_classes", [2, 3])
+def test_lda_predict_proba(n_classes):
+    def generate_dataset(n_samples, centers, covariances, random_state=None):
+        """Generate a multivariate normal data given some centers and
+        covariances"""
+        rng = check_random_state(random_state)
+        X = np.vstack([rng.multivariate_normal(mean, cov,
+                                               size=n_samples // len(centers))
+                       for mean, cov in zip(centers, covariances)])
+        y = np.hstack([[clazz] * (n_samples // len(centers))
+                       for clazz in range(len(centers))])
+        return X, y
+
+    blob_centers = np.array([[0, 0], [-10, 40], [-30, 30]])[:n_classes]
+    blob_stds = np.array([[[10, 10], [10, 100]]] * len(blob_centers))
+    X, y = generate_dataset(
+        n_samples=90000, centers=blob_centers, covariances=blob_stds,
+        random_state=42
+    )
+    lda = LinearDiscriminantAnalysis(solver='lsqr').fit(X, y)
```

Comment:
do we want to test this for the other solvers as well? how long does the test take given the amount of samples?

Reply:
Good point. 

The test takes about 0.42 s for each case, so 0.83 for each solver (binary and multi-class). 

The test passes for `solver=svd`, but fails when `solver=eigen`. 
This is probably related to #11727. 

@amueller Shall we only include `svd` and `lsqr` for now in the tests and take a note in that other PR to update the tests to also include `eigen` when a fix is submitted?

Reply:
OK, I think I have now fixed that other issue which was due to bad normalisation of the eigenvectors and was causing issues with probabilities for the `eigen` solver #11727 . I have updated the rst file accordingly.

Now all three solvers are tested in the non-regression test.